pkg/store/proxy_heap_test.go

```diff
@@ -22,6 +22,123 @@ func TestRmLabelsCornerCases(t *testing.T) {
 	}), labels.Labels{})
 }
 
+type testingRespSet struct {
+	bufferedResponse []*storepb.SeriesResponse
+	i                int
+}
+
+func (l *testingRespSet) Close() {}
+
+func (l *testingRespSet) At() *storepb.SeriesResponse {
+	return l.bufferedResponse[l.i]
+}
+
+func (l *testingRespSet) Next() bool {
+	l.i++
+	return l.i < len(l.bufferedResponse)
+}
+
+func (l *testingRespSet) StoreID() string {
+	return ""
+}
+
+func (l *testingRespSet) Labelset() string {
+	return ""
+}
+
+func (l *testingRespSet) Empty() bool {
+	return l.i >= len(l.bufferedResponse)
+}
+
+func TestProxyResponseHeapSort(t *testing.T) {
+	for _, tcase := range []struct {
+		input []respSet
+		exp   []*storepb.SeriesResponse
+	}{
+		// Different sorted response sets
```

Comment:
Maybe we can convert this description to a `name` field and use it in `t.Run(`. This way the IDE can help out with running individual test cases.

Reply:
Sounds good! Added